apps/coordinator/src/clients/transactions.ts

```diff
@@ -68,3 +104,79 @@ export const usePendingTransactions = () => {
     },
   };
 };
+
+export interface Coin {
+  prevTxId: string;
+  vout: number;
+  address: string;
+  value: string;
+  prevTxHex: string;
+  slice?: Slice;
+}
+
+// Service function for fetching transaction coins (spendable outputs from prev txs)
+const fetchTransactionCoins = async (
+  transaction: TransactionDetails,
+  client: BlockchainClient,
+) => {
+  const coins = new Map<string, Coin>();
+  for (const input of transaction.vin) {
+    const { txid, vout } = input;
+
+    // Skip if we don't have valid identifiers
+    if (!txid || vout === undefined) continue;
+    const coinId = `${txid}:${vout}`;
+
+    const [prevTxHex, prevTx] = await Promise.all([
+      client.getTransactionHex(txid),
+      client.getTransaction(txid),
+    ]);
+
+    if (!prevTx || !prevTxHex) {
+      throw new Error(
+        `Failed to fetch prev tx info for input ${coinId}. Are you sure the transaction was sent from this wallet?`,
+      );
+    }
+
+    const coinAddress = prevTx.vout[vout].scriptPubkeyAddress;
+    const coinValue = bitcoinsToSatoshis(prevTx.vout[vout].value.toString());
+
+    if (!coinAddress) {
+      throw new Error(`Failed to fetch coin address for input ${coinId}.`);
+    }
+
+    coins.set(coinAddress, {
+      prevTxId: txid,
+      vout,
+      address: coinAddress,
+      value: coinValue,
+      prevTxHex,
+    });
+  }
+  return coins;
+};
+
+/**
+ * @description Fetches all the coins for a given transaction.
+ * @param txid - The transaction ID to fetch coins from
+ * @returns The coins from the transaction
+ */
+export const useTransactionCoins = (txid: string) => {
```

Comment:
this is the query hook that will get the coin objects for a transaction, which requires querying for each txid of each input.